cea/utilities/solar_equations.py

```diff
@@ -150,21 +152,46 @@ def declination_degree(when, TY):
     ----------
     .. [1] http://pysolar.org/
     """
-    return 23.45 * math.sin((2 * math.pi / (TY)) * ((when.utctimetuple().tm_yday) - 81))
 
+    return 23.45 * np.vectorize(math.sin)((2 * math.pi / (TY)) * (day_date - 81))
 
-def get_hour_angle(when, longitude_deg):
-    solar_time = get_solar_time(longitude_deg, when)
+
+def get_hour_angle(longitude_deg, min_date, hour_date, day_date):
+    solar_time = get_solar_time(longitude_deg, min_date, hour_date, day_date)
     return 15 * (12 - solar_time)
 
 
-def get_solar_time(longitude_deg, when):
+def get_solar_time(longitude_deg, min_date, hour_date, day_date):
+    """
     "returns solar time in hours for the specified longitude and time," \
     " accurate only to the nearest minute."
-    when = when.utctimetuple()
-    return \
-        (
-            (when.tm_hour * 60 + when.tm_min + 4 * longitude_deg + equation_of_time(when.tm_yday))
-        /
-            60
-        )
\ No newline at end of file
+
+    Parameters
+    ----------
+    longitude_deg
+    min_date
+    hour_date
+    day_date
+
+    Returns
+    -------
+
+    """
+    solar_time_min = hour_date * 60 + min_date + 4 * longitude_deg + get_equation_of_time(day_date)
+
+    return solar_time_min/60
+
+def get_equation_of_time(day_date):
+    """
+
+    Parameters
+    ----------
+    day_date
+
+    Returns
+    -------
+
+    """
+    B = (day_date-1)*360/365
+    E = 229.2*(0.000075 + 0.001868*math.cos(B) - 0.032077*math.sin(B) - 0.014615*math.cos(2*B) - 0.04089*math.sin(2*B))
```

Comment:
we should collectively get in the habit of formatting expressions properly: there is a single space left and right of the operator. pycharm can format this for you and complains if you don't (squiggly lines are a dead give-away)